src/core/annotation.js

```diff
@@ -746,6 +739,37 @@ var LinkAnnotation = (function LinkAnnotationClosure() {
   return LinkAnnotation;
 })();
 
+var PopupAnnotation = (function PopupAnnotationClosure() {
+  function PopupAnnotation(parameters) {
+    Annotation.call(this, parameters);
+
+    this.data.annotationType = AnnotationType.POPUP;
+
+    var dict = parameters.dict;
+    if (!dict.has('Parent')) {
+      warn('Popup annotations without a parent annotation are not supported.');
+      return;
+    }
+
+    var parentItem = dict.get('Parent');
+    if (!parentItem) {
```

Comment:
Is it necessary to have two different cases for when the `Parent` entry isn't present, or is invalid?
Wouldn't it suffice to just replace both of them with e.g.
```javascript
...
var dict = parameters.dict;
var parentDict = dict.get('Parent');
if (!parentDict) {
  warn(...);
  return;
}
this.data.parentId = ...
```